Bio/Nexus/Trees.py

```diff
@@ -392,24 +414,54 @@ def is_compatible(self, tree2, threshold, strict=True):
         missing1 = set(tree2.get_taxa()) - set(self.get_taxa())
         if strict and (missing1 or missing2):
             if missing1:
-                print("Taxon/taxa %s is/are missing in tree %s" % (",".join(missing1), self.name))
+                print(
+                    "Taxon/taxa %s is/are missing in tree %s"
+                    % (",".join(missing1), self.name)
+                )
             if missing2:
-                print("Taxon/taxa %s is/are missing in tree %s" % (",".join(missing2), tree2.name))
+                print(
+                    "Taxon/taxa %s is/are missing in tree %s"
+                    % (",".join(missing2), tree2.name)
+                )
             raise TreeError("Can't compare trees with different taxon compositions.")
-        t1 = [(set(self.get_taxa(n)), self.node(n).data.support) for n in self.all_ids() if
-              self.node(n).succ and
-              (self.node(n).data and self.node(n).data.support and self.node(n).data.support >= threshold)]
-        t2 = [(set(tree2.get_taxa(n)), tree2.node(n).data.support) for n in tree2.all_ids() if
-              tree2.node(n).succ and
-              (tree2.node(n).data and tree2.node(n).data.support and tree2.node(n).data.support >= threshold)]
+        t1 = [
+            (set(self.get_taxa(n)), self.node(n).data.support)
+            for n in self.all_ids()
+            if self.node(n).succ
+            and (
+                self.node(n).data
+                and self.node(n).data.support
+                and self.node(n).data.support >= threshold
+            )
+        ]
+        t2 = [
+            (set(tree2.get_taxa(n)), tree2.node(n).data.support)
+            for n in tree2.all_ids()
+            if tree2.node(n).succ
+            and (
+                tree2.node(n).data
+                and tree2.node(n).data.support
+                and tree2.node(n).data.support >= threshold
+            )
```

Comment:
Would this look better without the redundant brackets? ``A and (B and C and D)`` is equivalent to ``A and B and C and D`` isn't it?

Reply:
Both amended, Thank you.

Reply:
That does look better :)